src/v/storage/log_manager.cc

```diff
@@ -435,6 +435,8 @@ ss::future<> log_manager::housekeeping_loop() {
 
         try {
             co_await housekeeping_scan(lowest_ts_to_retain());
+        } catch (const ss::gate_closed_exception& e) {
+            // stop() called
```

Comment:
We could instead `co_return;` here, or, rethrow and allow the [outer housekeeping loop](https://github.com/redpanda-data/redpanda/blob/46a4c2fc10e8ad726c4e63127518675274ddb2bf/src/v/storage/log_manager.cc#L296) to process it.

Reply:
I like these being logged. It usually is a signal of poor logic in the code. I'm fine with hiding it in a few places if you can prove (an if statement) that we are indeed shutting down. Otherwise, it is easy to hide a bug, potential bug, future bug. We don't even know where this gate closed exception is coming from. Maybe it is a bug somewhere in the compactor where it doesn't make progress for other reasons beside shutting down and we will be hiding it.

Reply:
Would something like [this](https://gist.github.com/WillemKauf/77244d64ba02ddeeaba806988e1ca9be) make the proposed changes better in your opinion?

>this gate closed exception is coming from

In the general case, I suppose this is correct- we can't quite reason about the exact gate that has been closed, but in the "shutting down" case, it's easy to reason that it's coming directly from the `log_manager`'s `_gate` since we don't check for `_gate.is_closed()` before calling `_gate.hold()` in a lot of housekeeping related functions in the `log_manager`.

Reply:
`ssx::is_shutdown_exception` could help here

Reply:
> `ssx::is_shutdown_exception` could help here

It's used in the [outer housekeeping loop](https://github.com/redpanda-data/redpanda/blob/46a4c2fc10e8ad726c4e63127518675274ddb2bf/src/v/storage/log_manager.cc#L296), but since we are catching any and all exceptions in `housekeeping_loop()`, we aren't propagating them up the call stack from this `try/catch` site.

Hence, one of the options is to rethrow here in case of a `ss::gate_closed_exception`.

Reply:
It can as easily be from `disk_log_impl::housekeeping`, no? I.e. after a topic is deleted. Propagating the exception would result in a bug in such case.

Reply:
Perhaps I misunderstand- I am proposing that one of the options is to rethrow a `ss::gate_closed_exception` from `disk_log_impl::housekeeping_loop()` [here](https://github.com/redpanda-data/redpanda/blob/46a4c2fc10e8ad726c4e63127518675274ddb2bf/src/v/storage/log_manager.cc#L439), and allow it to be caught in `disk_log_impl::housekeeping()` [here](https://github.com/redpanda-data/redpanda/blob/46a4c2fc10e8ad726c4e63127518675274ddb2bf/src/v/storage/log_manager.cc#L296-L308) (that is the propagation I am referring to, not propagating further from `disk_log_impl::housekeeping()`.)

[Here is another example diff of what this could look like](https://gist.github.com/WillemKauf/301c7b1c198fcc044f60d6f306c0c5a6)

Reply:
Oh fine. I didn't notice that `housekeeping` has a while loop around. It already takes care of that. Should work then.